### PR TITLE
Wrap Package

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR converts `wrap` into a completely installable package so that everything is available on the system even if the user deletes the source repo.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,13 +24,17 @@ else()
...
 # Install scripts
 install(FILES cmake/PybindWrap.cmake cmake/gtwrapConfig.cmake
               cmake/GtwrapUtils.cmake
         DESTINATION "${SCRIPT_INSTALL_DIR}/gtwrap")
 
+# Install wrapping scripts
+install(PROGRAMS scripts/pybind_wrap.py scripts/matlab_wrap.py TYPE BIN)
+
+# Install pybind11 directory
+install(DIRECTORY pybind11 TYPE LIB)
```
Comment (ProfFan) [inline CMakeLists.txt:36]: So, the final destination will be `<ROOT>/lib/gtwrap/pybind11`? Please list a final changeset when we run `make install`, where will the files be installed?

File: gtwrap/pybind_wrapper.py [hunk 2]
```diff
@@ -319,76 +318,3 @@ def wrap(self):
             wrapped_namespace=wrapped_namespace,
             boost_class_export=boost_class_export,
         )
-
-
-def main():
```
Comment (ProfFan) [inline gtwrap/pybind_wrapper.py:324]: I think GTSAM will need to be adapted for this, right?

File: CMakeLists.txt (hunk 1 above)
Comment (ProfFan) [reply CMakeLists.txt:36]: *ideally* here should be a comment describing this.

**Review by ProfFan — APPROVED** [review]
Great! Approving sans comments. Thanks Varun, this is going to be awesome!

File: CMakeLists.txt (hunk 1 above)
Comment (varunagrawal) [reply CMakeLists.txt:36]: ROOT will be whatever CMAKE_INSTALL_PREFIX is. And wouldn't the install_manifest.txt have that entire changeset?

File: gtwrap/pybind_wrapper.py (hunk 2 above)
Comment (varunagrawal) [reply gtwrap/pybind_wrapper.py:324]: I'll be checking for compatibility issues when I pull the lastest subtree. :)

File: CMakeLists.txt (hunk 1 above)
Comment (ProfFan) [reply CMakeLists.txt:36]: I know, I mean this would be friendly to new users (single point of information)

**Pull request merged**